### ISSUE TITLE
Email Based Confirmations Not Working if Current Session

### ISSUE THREAD

**Issue by thombruce** [thombruce/helvellyn#85]
_Aside: Also, confirmation is presently meaningless - we don't restrict access based on it in any way yet; add another issue for this._

**Describe the bug**
Unable to confirm a new user via the given URL in a confirmation email if they are already logged in. If they sign out before visiting the URL, they are confirmed as expected.

**To Reproduce**
1. Sign up as a new user
2. Remain signed in
3. Click link in confirmation email
4. Session is invalidated, user is not confirmed

**Expected behavior**
User should be confirmed regardless of current session state.

**Additional context**
Maybe this is a route permissions issue in Helvellyn.js... It sounds like one. Confirmation link is both a public and non-public link... it should function either way.

**Comment by thombruce** [comment]
And as of 658cd77bd53a2d182f103c5818fe31ac6f854d87 we now have the opposite problem...

User cannot confirm their account if they are logged out. Interesting...

This...

Y'know what...

### It's not what I thought...

There's another issue in that the session I obtain when I signup doesn't persist beyond a refresh...

- **Investigate cause of signup session being invalidated**

**Comment by thombruce** [comment]
`refresh-token` is undefined on signUp.

**Comment by thombruce** [comment]
Credible UsersController code performs these steps after successful user save:

```ruby
Credible::ConfirmationMailer.with(user: @user).confirmation_email.deliver_later
@session = ::Session.create(user: @user)
render :show, status: :created, location: @user
```

We know the email gets delivered, and that the attached user is valid and confirmation token works.

The `users/_user` partial called from User#show looks like this, contains this:

```ruby
if @session
  json.access_token @session.access_token
  json.jwt @session.access_token

  json.refresh_token @session.refresh_token
end
```

...and that Session should have been created just fine...

### The Issue is Helvellyn

Our partial is overridden there and contains no refresh_token:

```ruby
json.extract! user, :id, :name, :email, :created_at, :updated_at

json.access_token @session.access_token if @session

json.url user_url(user, format: :json)
```

It was altered a long time ago so we could include `name` in the output... but I think that's redundant now as we don't use this data soecificall.

I think we can just delete the partial in Helvellyn.

**Comment by thombruce** [comment]
### Scratch that!

We do need name for editing on the account page.

Better solution is maybe to move required items to `show` view and allow the `_user` partial to be overridden, to avoid this issue in future.

**Comment by thombruce** [comment]
Fixed by 5a46ba4e3b27d00261497a8162d46ab49ee93b57
